### PR TITLE
Fix p:d:T::copy_triangulation()

### PR DESCRIPTION
**Pull request by peterrum**
FYI @dravinsi

### CODE REVIEW

File: source/distributed/tria.cc [hunk 1]
```diff
@@ -3953,8 +3953,6 @@ namespace parallel
             other_distributed->coarse_cell_to_p4est_tree_permutation;
           p4est_tree_to_coarse_cell_permutation =
             other_distributed->p4est_tree_to_coarse_cell_permutation;
-          this->cell_attached_data = other_distributed->cell_attached_data;
-          this->data_transfer      = other_distributed->data_transfer;
```
[the remark this answers is not in the document]
Comment (bangerth) [reply source/distributed/tria.cc:3957]: In the `this` object, that is?

[the remark this answers is not in the document]
Comment (luca-heltai) [reply source/distributed/tria.cc:3957]: I think this is correct. tria.copy_triangulation(other_tria) expects tria (`this`) to be empty before calling. But the data transfer should not be copied.

[the remark this answers is not in the document]
Comment (marcfehling) [reply source/distributed/tria.cc:3957]: Yes, but we should do it in `p::d::T::clear`.

[the remark this answers is not in the document]
Comment (peterrum) [reply source/distributed/tria.cc:3957]: This is already done here: 

https://github.com/dealii/dealii/blob/81ffe76815c43ad5bd7ede10c920e206ebca0b1c/source/distributed/tria_base.cc#L679-L687

[the remark this answers is not in the document]
Comment (marcfehling) [reply source/distributed/tria.cc:3957]: Okay! I didn't check the base class :)

**Review by marcfehling — APPROVED** [review]

**Pull request merged**